diff --git a/index.js b/index.js
index 0ee8051..b19c1a1 100644
--- a/index.js
+++ b/index.js
@@ -51,6 +51,7 @@ class Peer extends stream.Duplex {
     this.trickle = opts.trickle !== undefined ? opts.trickle : true
     this.allowHalfTrickle = opts.allowHalfTrickle !== undefined ? opts.allowHalfTrickle : false
     this.iceCompleteTimeout = opts.iceCompleteTimeout || ICECOMPLETE_TIMEOUT
+    this.withDataChannel = opts.withDataChannel !== undefined ? opts.withDataChannel : true
 
     this.destroyed = false
     this.destroying = false
@@ -137,13 +138,15 @@ class Peer extends stream.Duplex {
     // - onfingerprintfailure
     // - onnegotiationneeded
 
-    if (this.initiator || this.channelNegotiated) {
-      this._setupData({
-        channel: this._pc.createDataChannel(this.channelName, this.channelConfig)
-      })
-    } else {
-      this._pc.ondatachannel = event => {
-        this._setupData(event)
+    if (this.withDataChannel) {
+      if (this.initiator || this.channelNegotiated) {
+        this._setupData({
+          channel: this._pc.createDataChannel(this.channelName, this.channelConfig)
+        })
+      } else {
+        this._pc.ondatachannel = event => {
+          this._setupData(event)
+        }
       }
     }
 
